--- config.py.ORIG	2017-04-18 10:00:00.000000000 -0400
+++ config.py	2017-04-18 10:00:24.000000000 -0400
@@ -246,7 +246,7 @@
             assert not os.path.isabs(package_name), ("package name should not be a absolute path, "
                                                      "to preserve croot during path joins")
             build_folders = sorted([build_folder for build_folder in get_build_folders(self.croot)
-                                if package_name in build_folder])
+                                if package_name + "_" in build_folder])
 
             if self.dirty and build_folders:
                 # Use the most recent build with matching recipe name
