--- config.py	2017-09-18 17:22:59.000000000 -0400
+++ config.py.new	2017-09-18 16:39:48.000000000 -0400
@@ -370,7 +370,7 @@
                                                      "to preserve croot during path joins")
             build_folders = sorted([os.path.basename(build_folder)
                             for build_folder in get_build_folders(self.croot)
-                            if os.path.basename(build_folder).startswith(package_name + "_")])
+                            if build_folder[:build_folder.rfind('_')] == package_name])
             if self.dirty and build_folders:
                 # Use the most recent build with matching recipe name
                 self._build_id = build_folders[-1]
